binding-hs/src/Client.hs

```diff
@@ -0,0 +1,192 @@
+{-# LANGUAGE DataKinds             #-}
+{-# LANGUAGE DeriveGeneric         #-}
+{-# LANGUAGE DuplicateRecordFields #-}
+{-# LANGUAGE MultiParamTypeClasses #-}
+{-# LANGUAGE OverloadedStrings     #-}
+{-# LANGUAGE TypeOperators         #-}
+
+module Client
+  ( envCreate
+  , envListAll
+  , envReset
+  , envStep
+  , envActionSpaceInfo
+  , envActionSpaceSample
+  , envActionSpaceContains
+  , envObservationSpaceInfo
+  , envMonitorStart
+  , envMonitorClose
+  , envClose
+  , upload
+  , shutdownServer
+  , EnvID (..)
+  , InstID (..)
+  , Environment (..)
+  , Observation (..)
+  , Step (..)
+  , Outcome (..)
+  , Info (..)
+  , Action (..)
+  , Monitor (..)
+  , Config (..)
+  ) where
+
+import           Data.Aeson          (FromJSON, Object (..), ToJSON)
+import qualified Data.HashMap.Strict as HM (HashMap (..))
+import           Data.Proxy          (Proxy (..))
+import qualified Data.Text           as T (Text)
```

Comment:
I really like your approach. I'll definitely do this despite having only a single file,
just in case if things change in the future.